torch/nn/parallel/distributed.py

```diff
@@ -521,6 +522,9 @@ class DistributedDataParallel(Module, Joinable):
         >>> net = torch.nn.parallel.DistributedDataParallel(model)
     """
 
+    # used to track whether the given thread is inside ddp forward for torchdynamo purposes
+    _tls_ctx = threading.local()
```

Comment:
This doesn't really track or enforce - it just stores. I would instead either store thread identifying info and fail if it stops matching, or, better yet, remove this and let the GIL handle this assumptions safely - even if multiple threads modify the `_tls_ctx`, they will do it only one at a time, right?

Reply:
I guess i wasn't sure i wanted to enforce users didn't use more than one thread as it could technically be valid, but at the same time I can't think of a good reason for it and the normal use case should be 1 py thread per process for DDP.  So I may just delete the tls thing altogether.